src/utils/swr/getRelevantCacheKeys.ts

```diff
@@ -0,0 +1,58 @@
+import { type Cache } from 'swr'
+import { withSWRKeyTags } from './withSWRKeyTags'
+
+type GetRelevantCacheKeysParameters = {
+  cache: Cache<unknown>
+  predicate: (tags: ReadonlyArray<string>) => boolean
+  withPrecedingOptimisticUpdate?: boolean
+}
+
+export function getRelevantCacheKeys({
+  cache,
+  predicate,
+  withPrecedingOptimisticUpdate,
+}: GetRelevantCacheKeysParameters) {
+  return Array.from(cache.keys())
+    .map((key) => {
+      const data = cache.get(key)
+      if (data && '_k' in data) {
+        const { _k: rawKey } = data
+
+        if (typeof rawKey === 'string' && rawKey.startsWith('$inf$')) {
+          const regexMatch = rawKey.match(/tags:@(.+),,/)
```

Comment:
This _probably_ doesn't matter, but typically regex compilation is non-trivially expensive and is one of the few things to bother moving outside of loops.

Reply:
nevermind looked up how this works in ts, disregard